heft-plugins/heft-api-extractor-plugin/src/ApiExtractorPlugin.ts

```diff
@@ -0,0 +1,225 @@
+  ): Promise<IApiExtractorConfigurationResult | undefined> {
+    // API Extractor provides an ExtractorConfig.tryLoadForFolder() API that will probe for api-extractor.json
+    // including support for rig.json.  However, Heft does not load the @microsoft/api-extractor package at all
+    // unless it sees a config/api-extractor.json file.  Thus we need to do our own lookup here.
+    const apiExtractorConfigurationFilePath: string | undefined =
+      await this._getApiExtractorConfigurationFilePathAsync(heftConfiguration);
+    if (!apiExtractorConfigurationFilePath) {
+      return undefined;
+    }
+
+    // Since the config file exists, we can assume that API Extractor is available. Attempt to resolve
+    // and import the package. If the resolution fails, a helpful error is thrown.
+    const apiExtractorPackage: typeof TApiExtractor = await this._getApiExtractorPackageAsync(
+      taskSession,
+      heftConfiguration
+    );
+    const apiExtractorConfigurationObject: TApiExtractor.IConfigFile =
+      apiExtractorPackage.ExtractorConfig.loadFile(apiExtractorConfigurationFilePath);
+
+    // Load the configuration file. Always load from scratch.
+    const apiExtractorConfiguration: TApiExtractor.ExtractorConfig =
+      apiExtractorPackage.ExtractorConfig.prepare({
+        ignoreMissingEntryPoint,
+        configObject: apiExtractorConfigurationObject,
+        configObjectFullPath: apiExtractorConfigurationFilePath,
+        packageJsonFullPath: `${heftConfiguration.buildFolder}/package.json`,
+        projectFolderLookupToken: heftConfiguration.buildFolder
+      });
+
+    return { apiExtractorPackage, apiExtractorConfiguration };
+  }
+
+  private async _getApiExtractorPackageAsync(
+    taskSession: IHeftTaskSession,
+    heftConfiguration: HeftConfiguration
+  ): Promise<typeof TApiExtractor> {
+    if (!this._apiExtractor) {
+      const apiExtractorPackagePath: string = await heftConfiguration.rigToolResolver.resolvePackageAsync(
+        '@microsoft/api-extractor',
+        taskSession.logger.terminal
+      );
+      this._apiExtractor = (await import(apiExtractorPackagePath)) as typeof TApiExtractor;
+    }
+    return this._apiExtractor;
+  }
+
+  private async _getApiExtractorTaskConfigurationAsync(
+    taskSession: IHeftTaskSession,
+    heftConfiguration: HeftConfiguration
+  ): Promise<IApiExtractorTaskConfiguration | undefined> {
+    if (!this._apiExtractorTaskConfigurationFileLoader) {
+      this._apiExtractorTaskConfigurationFileLoader = new ConfigurationFile<IApiExtractorTaskConfiguration>({
+        projectRelativeFilePath: TASK_CONFIG_RELATIVE_PATH,
+        jsonSchemaPath: TASK_CONFIG_SCHEMA_PATH
+      });
+    }
+
+    return await this._apiExtractorTaskConfigurationFileLoader.tryLoadConfigurationFileForProjectAsync(
+      taskSession.logger.terminal,
+      heftConfiguration.buildFolder,
+      heftConfiguration.rigConfig
+    );
+  }
+
+  private _includeOutputPathsInClean(
```

Comment:
Make sure the checked-in report file doesn't get included.

Reply:
Yep! There's a comment under the `apiReportEnabled` condition that explains why we leave `apiExtractorConfiguration.reportFilePath` as-is.